syntax = "proto3";

package calculator;

service Calculator {
  rpc Add (CalculationRequest) returns (CalculationResponse) {}
  rpc Subtract (CalculationRequest) returns (CalculationResponse) {}
  rpc Multiply (CalculationRequest) returns (CalculationResponse) {}
  rpc Divide (CalculationRequest) returns (CalculationResponse) {}
}

message CalculationRequest {
  float num1 = 1;
  float num2 = 2;
  string operation_id = 3;  // Unique identifier for tracking operations
}

message CalculationResponse {
  float result = 1;
  string operation_id = 2;
  bool success = 3;
  string error_message = 4;
}